### PR TITLE
master: support single master failover

### PR DESCRIPTION
**Pull request by amyangfei**
Part of #42 

This PR has several changes to the server master.

- Introduce an `initialized` field, before server loads information from metastore and initializes, this field is `false` and some server gRPC API will return `MasterNotReady` error.
- Support auto recovery of executor states in server master.

### CODE REVIEW

File: master/server.go [hunk 1]
```diff
@@ -166,6 +198,31 @@ func (s *Server) Start(ctx context.Context) (err error) {
 	if test.GlobalTestFlag {
 		return s.startForTest(ctx)
 	}
+
```
Comment (nongfushanquan) [inline master/server.go:202]: ToDo : ServerMaster should update the its address in etcd , so the executor can get it

File: master/server.go [hunk 2]
```diff
@@ -166,6 +198,31 @@ func (s *Server) Start(ctx context.Context) (err error) {
...
+	// rebuild states from existing meta if needed
+	err = s.resetExecutor(ctx)
+	if err != nil {
+		return err
+	}
+
+	// start background managers
+	s.executorManager.Start(ctx)
+	s.jobManager.Start(ctx)
```
Comment (nongfushanquan) [inline master/server.go:220]: the failover of jobmanager will be  implemented in the JobManager Start()  method ?

Comment (amyangfei) [reply master/server.go:220]: Will be implemented in another PR.

File: master/server.go (hunk 1 above)
Comment (amyangfei) [reply master/server.go:202]: It will be implemented in another PR.

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**